Animations/vendor/ffmpeg.cmake

```diff
@@ -0,0 +1,203 @@
+# FFMPEG build support
```

Comment:
I really appreciate you adding this in here, with that said it's not working for me for some reason :pain:.

I get an error that looks like this:

![tmp](https://user-images.githubusercontent.com/24964392/211222100-c3bc9d8a-9344-452c-af58-ea18e60842d0.PNG)

And a message in the build output that reads:

```batch
1>Performing configure step for 'ffmpeg'
1>cl.exe is unable to create an executable file.
1>If cl.exe is a cross-compiler, use the --enable-cross-compile option.
1>Only do this if you know what cross compiling means.
1>C compiler test failed.
1>
1>If you think configure made a mistake, make sure you are using the latest
1>version from Git.  If the latest version fails, report the problem to the
1>ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
1>Include the log file "ffbuild/config.log" produced by configure as this will help
1>solve the problem.
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(247,5): error MSB8066: Custom build for 'C:\dev\tmp\MathAnimation\build\CMakeFiles\c30ba50c530cdc578bdc7374cba7ba33\ffmpeg-configure.rule;C:\dev\tmp\MathAnimation\build\CMakeFiles\c30ba50c530cdc578bdc7374cba7ba33\ffmpeg-build.rule;C:\dev\tmp\MathAnimation\build\CMakeFiles\c30ba50c530cdc578bdc7374cba7ba33\ffmpeg-install.rule;C:\dev\tmp\MathAnimation\build\CMakeFiles\0809672f957b281836f397c0395ef8d7\ffmpeg-complete.rule;C:\dev\tmp\MathAnimation\build\CMakeFiles\3a5d0660df04986a880ec10876fdc81a\ffmpeg.rule' exited with code 1.
1>Done building project "ffmpeg.vcxproj" -- FAILED.
```

I'm not sure which environment you're running and if you're able to test these changes on Windows with MSVC, but if you aren't able to test the changes it will probably be pretty difficult to coordinate and get a working build.

Also, FFmpeg has some pretty strict rules when you're statically linking the libraries (because of mp4 patent issues). One of these rules is listing the exact commands you're using to build the binaries, and while I'm sure using CMake to define the rules is probably alright, I have no idea if they consider that alright. 

If we're not able to resolve this, it may be easier to leave the instructions in the README for now, and then link these manually by searching for the binaries kind of like I have it currently :)

Reply:
Also, it scares me a bit having complex build rules like this that I don't fully understand. Since I'll have to maintain this code and most likely change and extend this in the future, it will most likely be easier to have the end-user manually compile ffmpeg, or download some binaries. It's not very trivial to automate because of all the extra dependencies and it might be best to let the user handle this part themselves.

Of course I'm open to automating this, I just want to make sure it's rock solid before committing to it :)

Reply:
Yikes... I honestly expected it to fail. It's a bit of a decision which has to be taken here. Either we have a higher barrier of contributing to this project by forcing windows user to manually build ffmpeg or we try to take the burden of them.

This is more if an idea that went from "Maybe it's possible" to "that might actually work". I can totally understand if you don't want that complex code in the project.

I think it would make more sense to debug this in a more interactive session, as it seems to work both on Github CI and my local machine. My idea was for CMake to create a predefined, isolated environment. Obviously that didn't work :smile:. If you want to, we can try to debug this on Discord and see if we can get it to work reliably. Else we can also remove it and shift the responsibility back to the person wanting to build this project.

Reply:
[This repository](https://github.com/BtbN/FFmpeg-Builds) has been mentioned in a discussion on discord, maybe it makes sense to use that instead of building ffmpeg ourselves?

Reply:
So I have a working PR that's in progress that completely removes ffmpeg and switches to AV1. Because of that we can remove this entirely :)